Riot/Modules/Room/RoomViewController.m

```diff
@@ -5368,12 +5368,25 @@ - (void)removeMXSessionStateChangeNotificationsListener
         ];
     }
     
-    return @[
-        [self copyMenuItemWithEvent:event andCell:cell],
-        [self replyMenuItemWithEvent:event],
-        [self editMenuItemWithEvent:event],
-        [self moreMenuItemWithEvent:event andCell:cell]
-    ];
+    BOOL showMoreOption = ([event.wireType isEqual:kMXEventTypeStringRoomMember] && RiotSettings.shared.roomContextualMenuShowMoreOptionForStates) || (![event.wireType isEqual:kMXEventTypeStringRoomMember] && RiotSettings.shared.roomContextualMenuShowMoreOptionForMessages);
```

Comment:
There is plenty of state events, not only room members. You can use `event.isState` to cover all.